### PR TITLE
CSHARP-4094: Add 'comment' option to EstimatedDocumentCountOptions.

### PR DESCRIPTION
**Pull request by JamesKovacs**
CSHARP-4094 changes split from CSHARP-4095 (PR #789).

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs [hunk 1]
```diff
@@ -27,6 +27,7 @@ public void constructor_should_initialize_instance()
         {
             var result = new EstimatedDocumentCountOptions();
 
+            result.Comment.Should().BeNull();
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs:30]: We probably need to have similar to `AggregateOperationTests` comment tests in `EstimatedDocumentCountOperationTests` as well:
[Comment_get_and_set_should_work](https://github.com/mongodb/mongo-csharp-driver/blob/7308368ff2235d3166ff3b1748be9ec3e6e546b0/tests/MongoDB.Driver.Core.Tests/Core/Operations/AggregateOperationTests.cs#L190)
[CreateCommand_should_return_expected_result_when_Comment_is_set](https://github.com/mongodb/mongo-csharp-driver/blob/7308368ff2235d3166ff3b1748be9ec3e6e546b0/tests/MongoDB.Driver.Core.Tests/Core/Operations/AggregateOperationTests.cs#L405)

**Review by BorisDog — CHANGES_REQUESTED** [review]
Looks good, just some comment for unit tests.

File: tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs (hunk 1 above)
Comment (JamesKovacs) [reply tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs:30]: `EstimatedDocumentCountOperationsTests` already includes the test `Comment_get_and_set_should_work`. Added new test named `CreateCountOperation_should_return_expected_result_when_Comment_is_set`.

Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs:30]: Oh sorry, missed the `Comment_get_and_set_should_work`.
`CreateCountOperation_should_return_expected_result_when_Comment_is_set`: the intention was to check that comment field is sent on the wire. But that is probably `CountOperation` tests responsibility, and I don't see the comment tests there. I've probably missed that in the previous PR.
So we could test the actual wire representation here (like `CreateCommand_should_return_expected_result_when_ReadConcern_is_set` does) or add `CountOperation` tests  in another ticket.

**Review by BorisDog — APPROVED** [review]
LGTM, please notice the additional test comment.

File: tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs (hunk 1 above)
Comment (JamesKovacs) [reply tests/MongoDB.Driver.Tests/EstimatedDocumentCountOptionsTests.cs:30]: Added test to `CountOperationTests` to ensure wire representation of comment is correct. Testing it as part of `EstimatedDocumentCountOperationsTests` would have required digging through at least two private methods, which I felt would be fragile going forward.

**Pull request merged**